bundles/org.eclipse.e4.ui.widgets/src/org/eclipse/e4/ui/widgets/ImageBasedFrame.java

```diff
@@ -66,6 +66,10 @@ public ImageBasedFrame(Composite parent, Control toWrap, boolean vertical,
 			ImageBasedFrame frame = (ImageBasedFrame) event.widget;
 			frame.setCursor(null);
 		});
+		toWrap.addListener(SWT.ZoomChanged, event -> {
```

Comment:
I could guard it with if (getDisplay().isRescalingAtRuntime()) so the listener is only registered in this case

Reply:
If you do that and then some one changes the behavior (calls `setRescalingAtRuntime`) then the listener will be missing. If you want to guard then then always add listener but put the `if` **inside** the logic of the listener. I don't see a reason to guard it in the first place though since `ZoomChanged` events mean that the zoom changed and this logic needs to be executed, am I right? I mean only can always be conservative and guard against unnecessary events, but that would mean that one hides the problem (the problem being the **unnecessary** event).

Reply:
Yes, I totally agree. Thats why I didn't include the guard yet in the PR. I would leave it as is.